src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/ObserveWorldtubeSolution.hpp

```diff
@@ -118,6 +120,14 @@ struct ObserveWorldtubeSolution {
         }
       }();
       const auto current_time = db::get<::Tags::Time>(box);
+      Parallel::printf(MakeString{}
+                       << "Time: " << std::setprecision(16) << current_time
```

Comment:
You should add a tag for observation verbosity and wrap this in an `if` so people can turn this off if it gets annoying or is printing too much data.
```cpp
if (Parallel::get<Tags::WorldtubeVerbosity>(cache) >= ::Verbosity::Verbose) {
  ...
}
```

See the control system example in `src/ControlSystem/Tags/SystemTags.hpp{OptionTags.hpp}`

Reply:
[drive-by comment] Also, use printf properly with a format string.  If you want to use MakeString, you can just use the format "%s".  The reason for this is that if you do it the way you have and something you're printing happens to have a % in its output your code will crash.